webroot/src/components/ChangePassword/ChangePassword.ts

```diff
@@ -0,0 +1,192 @@
+                placeholder: computed(() => this.$t('common.newPassword')),
+                autocomplete: 'new-password',
+                validation: joiPassword
+                    .string()
+                    .min(12)
+                    .minOfSpecialCharacters(1)
+                    .minOfLowercase(1)
+                    .minOfUppercase(1)
+                    .minOfNumeric(1)
+                    .doesNotInclude([
+                        this.user.email,
+                        this.user.firstName,
+                        this.user.lastName,
+                        'password',
+                    ])
+                    .messages({
+                        ...this.joiMessages.string,
+                        ...this.joiMessages.password,
+                    }),
+            }),
+            confirmPassword: new FormInput({
+                id: 'confirm-password',
+                name: 'confirm-password',
+                label: computed(() => this.$t('common.confirmNewPassword')),
+                placeholder: computed(() => this.$t('common.confirmNewPassword')),
+                autocomplete: 'new-password',
+            }),
+            submit: new FormInput({
+                isSubmitInput: true,
+                id: 'submit-change-password',
+            }),
+        });
+        this.watchFormInputs(); // Important if you want automated form validation
+    }
+
+    async handleSubmit(): Promise<void> {
+        this.validateAll({ asTouched: true });
+        this.validateNewPassword();
+
+        if (this.isFormValid) {
+            this.startFormLoading();
+
+            await this.changePasswordRequest().catch((errMessage) => {
+                this.setError(errMessage);
+            });
+
+            if (!this.isFormError) {
+                this.isFormSuccessful = true;
+                this.updateFormSubmitSuccess(this.$t('common.success'));
+            }
+
+            this.endFormLoading();
+        }
+    }
+
+    validateNewPassword(): void {
+        const { confirmPassword: confirmPasswordInput } = this.formData;
+        const { newPassword, confirmPassword } = this.formValues;
+
+        if (confirmPasswordInput.isTouched && newPassword !== confirmPassword) {
+            confirmPasswordInput.errorMessage = this.$t('common.passwordsMustMatch');
+            confirmPasswordInput.isValid = false;
+        } else {
+            confirmPasswordInput.errorMessage = '';
+            confirmPasswordInput.isValid = true;
+        }
+    }
+
+    async changePasswordRequest(): Promise<void> {
+        const { currentPassword, newPassword } = this.formValues;
+        const requestData = JSON.stringify({
+            AccessToken: this.authToken,
+            PreviousPassword: currentPassword,
+            ProposedPassword: newPassword,
+        });
+        const { cognitoRegion } = this.$envConfig;
+
+        await axios.post(`https://cognito-idp.${cognitoRegion}.amazonaws.com/`, requestData, {
```

Comment:
Do we want to abstract this out to the network layer? Im assuming you didn't because it essentially adds more work for no upside? It just seems to be against the regular pattern. We also have no mock fallback functionality this way.